tests/test_data/custom_phenotypes/test_custom_phen_conversion.py

```diff
@@ -0,0 +1,56 @@
+import os
+import csv
+import pytest
+from tretools.utility.custom_phen_conversion import process_custom_codelist
+
+@pytest.fixture
+def input_file_path(tmp_path):
+    input_file = tmp_path / "input_file.csv"
+    content = """phenotype,code,term,name,comment
+GNH0001_AtrialFibrillationFlutter,I480,ICD10,Paroxysmal atrial fibrillation,
+GNH0001_AtrialFibrillationFlutter,I481,ICD10,Persistent atrial fibrillation,
+GNH0001_AtrialFibrillationFlutter,5370000,SNOMED,Atrial flutter,
+GNH0001_AtrialFibrillationFlutter,49436004,SNOMED,Atrial fibrillation,
+GNH0002_CoronaryArteryDisease_narrow,K403,OPCS4,Saphenous vein graft replacement of three coronary arteries,
+GNH0002_CoronaryArteryDisease_narrow,K404,OPCS4,Saphenous vein graft replacement of four or more coronary arteries,
+GNH0002_CoronaryArteryDisease_narrow,K411,OPCS4,Autograft replacement of one coronary artery NEC,
+Primary_malignancy_mesothelioma,115232000,SNOMED,"[M]Mesothelioma, unspecified",
+Primary_malignancy_mesothelioma,254645002,SNOMED,Malignant mesothelioma of pleura,
+Primary_malignancy_oesophageal,C15,ICD10,Malignant neoplasm of oesophagus,
+"""
+    input_file.write_text(content)
+    return input_file
+
+def test_process_custom_codelist(input_file_path, tmp_path):
+    output_directory = tmp_path
+    output_prefix = "customs"
+
+    process_custom_codelist(input_file_path, output_directory, output_prefix)
```

Comment:
ideally we want to test both the inner function (although it should be pulled out as a standalone function) and the function that calls these functions. 

plus tests for the exception.

Reply:
I intentionally defined a separate function for reading so that we can test the reading separately. No nested function any more. Also a test is written for the exception.